source/slang/slang-ast-type.cpp

```diff
@@ -387,15 +387,19 @@ Val* ArrayExpressionType::_substituteImplOverride(ASTBuilder* astBuilder, Substi
 {
     int diff = 0;
     auto elementType = as<Type>(baseType->substituteImpl(astBuilder, subst, &diff));
-    auto arrlen = as<IntVal>(arrayLength->substituteImpl(astBuilder, subst, &diff));
-    SLANG_ASSERT(arrlen);
+    IntVal* newArrayLength = nullptr;
+    if (arrayLength)
+    {
+        newArrayLength = as<IntVal>(arrayLength->substituteImpl(astBuilder, subst, &diff));
+        SLANG_ASSERT(newArrayLength);
+    }
     if (diff)
     {
         *ioDiff = 1;
```

Comment:
Is there a reason to not increment, or add the diff?

Reply:
Shouldn't matter. This is just used as a `bool`.